src/rpc/blockchain.cpp

```diff
@@ -183,20 +183,41 @@ UniValue blockToJSON(const CBlock& block, const CBlockIndex* blockindex, bool fP
 
 UniValue dumpcontracts(const UniValue& params, bool fHelp)
 {
-    if (fHelp || params.size() > 4)
-        throw runtime_error(
-                "dumpcontracts <contract_type> <file> <low height> <high height>\n"
+    if (fHelp || params.size() < 2 || params.size() > 5) {
+        std::stringstream help;
+
+        help << "dumpcontracts <contract_type> <file> [txids only] [low height] [high height]\n"
                 "\n"
-                "<contract_type> Contract type to gather data from."
+                "<contract_type> Contract type for which to gather data. Use \"*\" for all.\n"
```

Comment:
should this be "from which"?

Reply:
I think the original wording of `Contract type to gather data from` is probably better here for the first sentence. `for which` and `from which` both sound awkward here to me

Reply:
I don't like "from which", because you are not gathering data from a contract type. You are gathering data from contracts for a particular contract type.

Reply:
And... you are not supposed to end with a preposition.

Reply:
But... we don't have to be perfectly grammatically correct. If you both like the original wording, I will change it back! :)